src/data/h1/invader.js

```diff
@@ -0,0 +1,39 @@
+/* This file contains utility functions for working with Invader's
+ * tag definition JSON files
+ */
+
+//for the given struct, get all structs which comprise it
+function expandStructs(struct, structs) {
+  return struct.fields
+    .filter(field => field.type == "TagReflexive" &&
+      field.struct != "PredictedResource"
+    )
+    .map(field => structs[field.struct]);
+}
+
+//get all tag names (e.g. sound_looping) referenced by the given struct (e.g. SoundScenery)
+function getDirectReferencedTagNames(structName, structs) {
```

Comment:
This was previously within the `tag.js` page template. I was able to simplify it a bit now that tags have direct references to their parents and it's easier to build a multi-level references list without relying on invader structs for that.

Reply:
:D